gffquant/counters/alignment_counter.py

```
@@ -25,28 +34,225 @@ def get_increment(self, n_aln, increment):
+            uniq_scaling_factor, ambig_scaling_factor,            
+        )
+
+        # return count sums and scaling factors
+        return count_sums, uniq_scaling_factor, ambig_scaling_factor
+    
+    def group_gene_count_matrix(self, refmgr):
+        ggroup_index = {}
+        for key, key_index in self.index.items():
+            ref = (refmgr.get(key[0] if isinstance(key, tuple) else key))[0]
+            ref_tokens = ref.split(".")
+            gene_id, ggroup_id = ".".join(ref_tokens[:-1]), ref_tokens[-1]
```

Ruff (0.8.2)

Local variable `gene_id` is assigned to but never used

Remove assignment to unused variable `gene_id`

(F841)